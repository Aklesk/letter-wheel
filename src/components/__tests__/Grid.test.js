import React from 'react';
import { mount } from 'enzyme';
import { View as AnimatableView } from 'react-native-animatable';
import { Text, TouchableOpacity, Vibration } from 'react-native';

import Grid from '../Grid';

Vibration.vibrate = jest.fn();

describe('Grid component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    const render = mount((
      <Grid />
    ));
    expect(render).toMatchSnapshot();
    render.setProps({
      gridEntries: [
        { letter: 'a', selected: false },
        { letter: 's', selected: false },
        { letter: 'd', selected: true },
        { letter: 'f', selected: true },
        { letter: 'g', selected: false },
        { letter: 'h', selected: false },
        { letter: 'j', selected: false },
        { letter: 'k', selected: false },
        { letter: 'l', selected: true },
      ],
    });
    expect(render).toMatchSnapshot();
  });

  it('provides submitWord function that triggers animation', () => {
    const render = mount((
      <Grid
        submitWord={jest.fn()}
        triggerAnimation={jest.fn()}
      />
    ));
    render.instance().submitWord();
    expect(render.props().submitWord).toHaveBeenCalledTimes(1);
    expect(render.props().triggerAnimation).toHaveBeenCalledTimes(0);

    // Incorrect word
    jest.clearAllMocks();
    render.props().submitWord.mockReturnValueOnce(false);
    render.instance().submitWord();
    expect(render.props().submitWord).toHaveBeenCalledTimes(1);
    expect(render.props().triggerAnimation).toHaveBeenCalledTimes(1);
    expect(render.props().triggerAnimation).toHaveBeenCalledWith('incorrect');

    // Correct word
    jest.clearAllMocks();
    render.props().submitWord.mockReturnValueOnce(true);
    render.instance().submitWord();
    expect(render.props().submitWord).toHaveBeenCalledTimes(1);
    expect(render.props().triggerAnimation).toHaveBeenCalledTimes(1);
    expect(render.props().triggerAnimation).toHaveBeenCalledWith('correct');
  });

  it('triggers animations when the animationState prop changes', () => {
    const render = mount((
      <Grid
        clearAnimation={jest.fn()}
        submitWord={jest.fn()}
        triggerAnimation={jest.fn()}
      />
    ));

    let animatable = render.find(AnimatableView).at(0);
    expect(animatable.props().animation).toEqual({ 0: {}, 1: {} });
    expect(animatable.props().duration).toEqual(0);
    expect(animatable.props().onAnimationEnd).toEqual(render.props().clearAnimation);

    jest.clearAllMocks();
    render.setProps({ animationState: 'correct' });
    animatable = render.find(AnimatableView).at(0);
    expect(animatable.props().animation).toEqual({
      0.00: { scale: 1 },
      0.05: { scale: 1 },
      0.30: { scale: 1.05 },
      0.70: { scale: 0.95 },
      0.95: { scale: 1 },
      1.00: { scale: 1 },
    });
    expect(animatable.props().duration).toEqual(1000);
    expect(animatable.props().onAnimationEnd).toEqual(render.props().clearAnimation);

    jest.clearAllMocks();
    render.setProps({ animationState: 'incorrect' });
    animatable = render.find(AnimatableView).at(0);
    expect(animatable.props().animation).toEqual({
      0.000: { translateX: 0 },
      0.250: { translateX: -6 },
      0.375: { translateX: 6 },
      0.500: { translateX: -6 },
      0.625: { translateX: 6 },
      0.750: { translateX: -6 },
      1.000: { translateX: 0 },
    });
    expect(animatable.props().duration).toEqual(500);
    expect(animatable.props().onAnimationEnd).toEqual(render.props().clearAnimation);

    jest.clearAllMocks();
    render.setProps({ animationState: '' });
    animatable = render.find(AnimatableView).at(0);
    expect(animatable.props().animation).toEqual({ 0: {}, 1: {} });
    expect(animatable.props().duration).toEqual(0);
    expect(animatable.props().onAnimationEnd).toEqual(render.props().clearAnimation);
  });

  it('toggles selected grid item on touch', () => {
    const render = mount((
      <Grid
        clearAnimation={jest.fn()}
        submitWord={jest.fn()}
        toggleSelected={jest.fn()}
        triggerAnimation={jest.fn()}
      />
    ));

    expect(render.props().toggleSelected).toHaveBeenCalledTimes(0);
    render.find(TouchableOpacity).at(0).props().onPress();
    expect(render.props().toggleSelected).toHaveBeenCalledTimes(1);
    expect(render.props().toggleSelected).toHaveBeenCalledWith('0');

    jest.clearAllMocks();
    render.find(TouchableOpacity).at(0).props().onPress();
    expect(render.props().toggleSelected).toHaveBeenCalledTimes(1);
    expect(render.props().toggleSelected).toHaveBeenCalledWith('0');

    jest.clearAllMocks();
    render.find(TouchableOpacity).at(4).props().onPress();
    expect(render.props().toggleSelected).toHaveBeenCalledTimes(1);
    expect(render.props().toggleSelected).toHaveBeenCalledWith('4');
  });

  it('submits and selects if needed on long press', () => {
    const render = mount((
      <Grid
        clearAnimation={jest.fn()}
        gridEntries={[
          { letter: 'a', selected: false },
          { letter: 's', selected: false },
          { letter: 'd', selected: true },
          { letter: 'f', selected: true },
          { letter: 'g', selected: false },
          { letter: 'h', selected: false },
          { letter: 'j', selected: false },
          { letter: 'k', selected: false },
          { letter: 'l', selected: true },
        ]}
        submitWord={jest.fn()}
        toggleSelected={jest.fn()}
        triggerAnimation={jest.fn()}
      />
    ));

    jest.spyOn(render.instance(), 'submitWord');

    expect(render.props().toggleSelected).toHaveBeenCalledTimes(0);
    expect(Vibration.vibrate).toHaveBeenCalledTimes(0);
    render.find(TouchableOpacity).at(0).props().onLongPress();
    expect(Vibration.vibrate).toHaveBeenCalledTimes(1);
    expect(Vibration.vibrate).toHaveBeenCalledWith(100);
    expect(render.props().toggleSelected).toHaveBeenCalledTimes(1);
    expect(render.props().toggleSelected).toHaveBeenCalledWith('0');
    expect(render.props().submitWord).toHaveBeenCalledTimes(1);

    jest.clearAllMocks();
    expect(render.props().toggleSelected).toHaveBeenCalledTimes(0);
    expect(Vibration.vibrate).toHaveBeenCalledTimes(0);
    render.find(TouchableOpacity).at(2).props().onLongPress();
    expect(Vibration.vibrate).toHaveBeenCalledTimes(1);
    expect(Vibration.vibrate).toHaveBeenCalledWith(100);
    expect(render.props().toggleSelected).toHaveBeenCalledTimes(0);
    expect(render.props().submitWord).toHaveBeenCalledTimes(1);

    jest.clearAllMocks();
    expect(render.props().toggleSelected).toHaveBeenCalledTimes(0);
    expect(Vibration.vibrate).toHaveBeenCalledTimes(0);
    render.find(TouchableOpacity).at(7).props().onLongPress();
    expect(Vibration.vibrate).toHaveBeenCalledTimes(1);
    expect(Vibration.vibrate).toHaveBeenCalledWith(100);
    expect(render.props().toggleSelected).toHaveBeenCalledTimes(1);
    expect(render.props().toggleSelected).toHaveBeenCalledWith('7');
    expect(render.props().submitWord).toHaveBeenCalledTimes(1);
  });

  it('provides a clearOne function', () => {
    const render = mount((
      <Grid
        clearSelected={jest.fn()}
      />
    ));
    expect(render.instance().clearOne).toEqual(expect.any(Function));
    expect(render.instance().clearOne()).toBeUndefined();
    expect(render.props().clearSelected).toHaveBeenCalledTimes(1);
    expect(render.props().clearSelected).toHaveBeenCalledWith(1);
  });

  it('provides a clearAll function', () => {
    const render = mount((
      <Grid
        clearSelected={jest.fn()}
      />
    ));
    expect(render.instance().clearAll).toEqual(expect.any(Function));
    expect(render.instance().clearAll()).toBeUndefined();
    expect(render.props().clearSelected).toHaveBeenCalledTimes(1);
    expect(render.props().clearSelected).toHaveBeenCalledWith(9);
    expect(Vibration.vibrate).toHaveBeenCalledTimes(1);
    expect(Vibration.vibrate).toHaveBeenCalledWith(100);
  });

  it('provides a display of the selected letters', () => {
    const render = mount((
      <Grid
        selectedLetters="fish"
      />
    ));
    expect(render.find(Text).find({ children: 'fish' })).toHaveLength(2);
  });

  it('provides a backspace button with a short and long press action', () => {
    const render = mount((
      <Grid />
    ));
    expect(render.find({
      onPress: render.instance().clearOne,
      onLongPress: render.instance().clearAll,
    })).toHaveLength(1);
  });
});
