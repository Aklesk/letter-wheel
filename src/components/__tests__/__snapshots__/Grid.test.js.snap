// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid component matches the snapshot 1`] = `
<Grid
  animationState=""
  clearAnimation={[Function]}
  clearSelected={[Function]}
  gridEntries={
    Array [
      Object {
        "letter": "",
        "selected": false,
      },
      Object {
        "letter": "",
        "selected": false,
      },
      Object {
        "letter": "",
        "selected": false,
      },
      Object {
        "letter": "",
        "selected": false,
      },
      Object {
        "letter": "",
        "selected": false,
      },
      Object {
        "letter": "",
        "selected": false,
      },
      Object {
        "letter": "",
        "selected": false,
      },
      Object {
        "letter": "",
        "selected": false,
      },
      Object {
        "letter": "",
        "selected": false,
      },
    ]
  }
  selectedLetters=""
  submitWord={[Function]}
  toggleSelected={[Function]}
  triggerAnimation={[Function]}
>
  <withAnimatable(View)
    animation={
      Object {
        "0": Object {},
        "1": Object {},
      }
    }
    delay={0}
    direction="normal"
    duration={0}
    iterationCount={1}
    onAnimationBegin={[Function]}
    onAnimationEnd={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "justifyContent": "center",
      }
    }
    useNativeDriver={false}
  >
    <AnimatedComponent
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "justifyContent": "center",
          },
          Object {},
          Object {},
        ]
      }
    >
      <View>
        <react-native-mock>
          <ImageBackground
            source={
              Object {
                "testUri": "src/images/gridBackground.png",
              }
            }
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "height": null,
                "justifyContent": "center",
                "paddingBottom": 2,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 8,
                "width": null,
              }
            }
          >
            <react-native-mock>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 300,
                    "justifyContent": "center",
                    "width": 300,
                  }
                }
              >
                <react-native-mock>
                  <View
                    key="012"
                    style={
                      Object {
                        "flex": 0.3333333333333333,
                        "flexDirection": "row",
                        "margin": 0,
                      }
                    }
                  >
                    <react-native-mock>
                      <TouchableOpacity
                        key="gridItem0"
                        onLongPress={[Function]}
                        onPress={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "borderColor": "#000000",
                            "borderWidth": 2,
                            "flex": 0.3333333333333333,
                            "justifyContent": "center",
                            "margin": 0,
                          }
                        }
                      >
                        <react-native-mock>
                          <Text
                            style={
                              Object {
                                "color": "#000",
                                "fontSize": 60,
                              }
                            }
                          >
                            <react-native-mock />
                          </Text>
                        </react-native-mock>
                      </TouchableOpacity>
                      <TouchableOpacity
                        key="gridItem1"
                        onLongPress={[Function]}
                        onPress={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "borderColor": "#000000",
                            "borderWidth": 2,
                            "flex": 0.3333333333333333,
                            "justifyContent": "center",
                            "margin": 0,
                          }
                        }
                      >
                        <react-native-mock>
                          <Text
                            style={
                              Object {
                                "color": "#000",
                                "fontSize": 60,
                              }
                            }
                          >
                            <react-native-mock />
                          </Text>
                        </react-native-mock>
                      </TouchableOpacity>
                      <TouchableOpacity
                        key="gridItem2"
                        onLongPress={[Function]}
                        onPress={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "borderColor": "#000000",
                            "borderWidth": 2,
                            "flex": 0.3333333333333333,
                            "justifyContent": "center",
                            "margin": 0,
                          }
                        }
                      >
                        <react-native-mock>
                          <Text
                            style={
                              Object {
                                "color": "#000",
                                "fontSize": 60,
                              }
                            }
                          >
                            <react-native-mock />
                          </Text>
                        </react-native-mock>
                      </TouchableOpacity>
                    </react-native-mock>
                  </View>
                  <View
                    key="345"
                    style={
                      Object {
                        "flex": 0.3333333333333333,
                        "flexDirection": "row",
                        "margin": 0,
                      }
                    }
                  >
                    <react-native-mock>
                      <TouchableOpacity
                        key="gridItem3"
                        onLongPress={[Function]}
                        onPress={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "borderColor": "#000000",
                            "borderWidth": 2,
                            "flex": 0.3333333333333333,
                            "justifyContent": "center",
                            "margin": 0,
                          }
                        }
                      >
                        <react-native-mock>
                          <Text
                            style={
                              Object {
                                "color": "#000",
                                "fontSize": 60,
                              }
                            }
                          >
                            <react-native-mock />
                          </Text>
                        </react-native-mock>
                      </TouchableOpacity>
                      <TouchableOpacity
                        key="gridItem4"
                        onLongPress={[Function]}
                        onPress={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#aaa",
                            "borderColor": "#000000",
                            "borderWidth": 2,
                            "flex": 0.3333333333333333,
                            "justifyContent": "center",
                            "margin": 0,
                          }
                        }
                      >
                        <react-native-mock>
                          <Text
                            style={
                              Object {
                                "color": "#000",
                                "fontSize": 60,
                              }
                            }
                          >
                            <react-native-mock />
                          </Text>
                        </react-native-mock>
                      </TouchableOpacity>
                      <TouchableOpacity
                        key="gridItem5"
                        onLongPress={[Function]}
                        onPress={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "borderColor": "#000000",
                            "borderWidth": 2,
                            "flex": 0.3333333333333333,
                            "justifyContent": "center",
                            "margin": 0,
                          }
                        }
                      >
                        <react-native-mock>
                          <Text
                            style={
                              Object {
                                "color": "#000",
                                "fontSize": 60,
                              }
                            }
                          >
                            <react-native-mock />
                          </Text>
                        </react-native-mock>
                      </TouchableOpacity>
                    </react-native-mock>
                  </View>
                  <View
                    key="678"
                    style={
                      Object {
                        "flex": 0.3333333333333333,
                        "flexDirection": "row",
                        "margin": 0,
                      }
                    }
                  >
                    <react-native-mock>
                      <TouchableOpacity
                        key="gridItem6"
                        onLongPress={[Function]}
                        onPress={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "borderColor": "#000000",
                            "borderWidth": 2,
                            "flex": 0.3333333333333333,
                            "justifyContent": "center",
                            "margin": 0,
                          }
                        }
                      >
                        <react-native-mock>
                          <Text
                            style={
                              Object {
                                "color": "#000",
                                "fontSize": 60,
                              }
                            }
                          >
                            <react-native-mock />
                          </Text>
                        </react-native-mock>
                      </TouchableOpacity>
                      <TouchableOpacity
                        key="gridItem7"
                        onLongPress={[Function]}
                        onPress={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "borderColor": "#000000",
                            "borderWidth": 2,
                            "flex": 0.3333333333333333,
                            "justifyContent": "center",
                            "margin": 0,
                          }
                        }
                      >
                        <react-native-mock>
                          <Text
                            style={
                              Object {
                                "color": "#000",
                                "fontSize": 60,
                              }
                            }
                          >
                            <react-native-mock />
                          </Text>
                        </react-native-mock>
                      </TouchableOpacity>
                      <TouchableOpacity
                        key="gridItem8"
                        onLongPress={[Function]}
                        onPress={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "borderColor": "#000000",
                            "borderWidth": 2,
                            "flex": 0.3333333333333333,
                            "justifyContent": "center",
                            "margin": 0,
                          }
                        }
                      >
                        <react-native-mock>
                          <Text
                            style={
                              Object {
                                "color": "#000",
                                "fontSize": 60,
                              }
                            }
                          >
                            <react-native-mock />
                          </Text>
                        </react-native-mock>
                      </TouchableOpacity>
                    </react-native-mock>
                  </View>
                </react-native-mock>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "height": 50,
                    "justifyContent": "flex-start",
                    "marginHorizontal": 10,
                  }
                }
              >
                <react-native-mock>
                  <View
                    style={
                      Object {
                        "borderBottomColor": "#444",
                        "borderBottomWidth": 1,
                        "flex": 1,
                        "height": 30,
                        "marginLeft": 20,
                      }
                    }
                  >
                    <react-native-mock>
                      <Text
                        style={
                          Object {
                            "color": "#000",
                            "fontSize": 20,
                          }
                        }
                      >
                        <react-native-mock />
                      </Text>
                    </react-native-mock>
                  </View>
                  <View>
                    <react-native-mock>
                      <TouchableOpacity
                        onLongPress={[Function]}
                        onPress={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "borderRadius": 8,
                            "height": 50,
                            "justifyContent": "center",
                            "width": 60,
                          }
                        }
                      >
                        <react-native-mock>
                          <Icon
                            allowFontScaling={false}
                            name="md-backspace"
                            size={35}
                          >
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": undefined,
                                    "fontSize": 35,
                                  },
                                  undefined,
                                  Object {
                                    "fontFamily": "Ionicons",
                                    "fontStyle": "normal",
                                    "fontWeight": "normal",
                                  },
                                ]
                              }
                            >
                              <react-native-mock>
                                
                              </react-native-mock>
                            </Text>
                          </Icon>
                        </react-native-mock>
                      </TouchableOpacity>
                    </react-native-mock>
                  </View>
                </react-native-mock>
              </View>
            </react-native-mock>
          </ImageBackground>
        </react-native-mock>
      </View>
    </AnimatedComponent>
  </withAnimatable(View)>
</Grid>
`;

exports[`Grid component matches the snapshot 2`] = `
<Grid
  animationState=""
  clearAnimation={[Function]}
  clearSelected={[Function]}
  gridEntries={
    Array [
      Object {
        "letter": "a",
        "selected": false,
      },
      Object {
        "letter": "s",
        "selected": false,
      },
      Object {
        "letter": "d",
        "selected": true,
      },
      Object {
        "letter": "f",
        "selected": true,
      },
      Object {
        "letter": "g",
        "selected": false,
      },
      Object {
        "letter": "h",
        "selected": false,
      },
      Object {
        "letter": "j",
        "selected": false,
      },
      Object {
        "letter": "k",
        "selected": false,
      },
      Object {
        "letter": "l",
        "selected": true,
      },
    ]
  }
  selectedLetters=""
  submitWord={[Function]}
  toggleSelected={[Function]}
  triggerAnimation={[Function]}
>
  <withAnimatable(View)
    animation={
      Object {
        "0": Object {},
        "1": Object {},
      }
    }
    delay={0}
    direction="normal"
    duration={0}
    iterationCount={1}
    onAnimationBegin={[Function]}
    onAnimationEnd={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "justifyContent": "center",
      }
    }
    useNativeDriver={false}
  >
    <AnimatedComponent
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "justifyContent": "center",
          },
          Object {},
          Object {},
        ]
      }
    >
      <View>
        <react-native-mock>
          <ImageBackground
            source={
              Object {
                "testUri": "src/images/gridBackground.png",
              }
            }
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "height": null,
                "justifyContent": "center",
                "paddingBottom": 2,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 8,
                "width": null,
              }
            }
          >
            <react-native-mock>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 300,
                    "justifyContent": "center",
                    "width": 300,
                  }
                }
              >
                <react-native-mock>
                  <View
                    key="012"
                    style={
                      Object {
                        "flex": 0.3333333333333333,
                        "flexDirection": "row",
                        "margin": 0,
                      }
                    }
                  >
                    <react-native-mock>
                      <TouchableOpacity
                        key="gridItem0"
                        onLongPress={[Function]}
                        onPress={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "borderColor": "#000000",
                            "borderWidth": 2,
                            "flex": 0.3333333333333333,
                            "justifyContent": "center",
                            "margin": 0,
                          }
                        }
                      >
                        <react-native-mock>
                          <Text
                            style={
                              Object {
                                "color": "#000",
                                "fontSize": 60,
                              }
                            }
                          >
                            <react-native-mock>
                              A
                            </react-native-mock>
                          </Text>
                        </react-native-mock>
                      </TouchableOpacity>
                      <TouchableOpacity
                        key="gridItem1"
                        onLongPress={[Function]}
                        onPress={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "borderColor": "#000000",
                            "borderWidth": 2,
                            "flex": 0.3333333333333333,
                            "justifyContent": "center",
                            "margin": 0,
                          }
                        }
                      >
                        <react-native-mock>
                          <Text
                            style={
                              Object {
                                "color": "#000",
                                "fontSize": 60,
                              }
                            }
                          >
                            <react-native-mock>
                              S
                            </react-native-mock>
                          </Text>
                        </react-native-mock>
                      </TouchableOpacity>
                      <TouchableOpacity
                        key="gridItem2"
                        onLongPress={[Function]}
                        onPress={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#ffa",
                            "borderColor": "#000000",
                            "borderWidth": 2,
                            "flex": 0.3333333333333333,
                            "justifyContent": "center",
                            "margin": 0,
                          }
                        }
                      >
                        <react-native-mock>
                          <Text
                            style={
                              Object {
                                "color": "#000",
                                "fontSize": 60,
                              }
                            }
                          >
                            <react-native-mock>
                              D
                            </react-native-mock>
                          </Text>
                        </react-native-mock>
                      </TouchableOpacity>
                    </react-native-mock>
                  </View>
                  <View
                    key="345"
                    style={
                      Object {
                        "flex": 0.3333333333333333,
                        "flexDirection": "row",
                        "margin": 0,
                      }
                    }
                  >
                    <react-native-mock>
                      <TouchableOpacity
                        key="gridItem3"
                        onLongPress={[Function]}
                        onPress={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#ffa",
                            "borderColor": "#000000",
                            "borderWidth": 2,
                            "flex": 0.3333333333333333,
                            "justifyContent": "center",
                            "margin": 0,
                          }
                        }
                      >
                        <react-native-mock>
                          <Text
                            style={
                              Object {
                                "color": "#000",
                                "fontSize": 60,
                              }
                            }
                          >
                            <react-native-mock>
                              F
                            </react-native-mock>
                          </Text>
                        </react-native-mock>
                      </TouchableOpacity>
                      <TouchableOpacity
                        key="gridItem4"
                        onLongPress={[Function]}
                        onPress={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#aaa",
                            "borderColor": "#000000",
                            "borderWidth": 2,
                            "flex": 0.3333333333333333,
                            "justifyContent": "center",
                            "margin": 0,
                          }
                        }
                      >
                        <react-native-mock>
                          <Text
                            style={
                              Object {
                                "color": "#000",
                                "fontSize": 60,
                              }
                            }
                          >
                            <react-native-mock>
                              G
                            </react-native-mock>
                          </Text>
                        </react-native-mock>
                      </TouchableOpacity>
                      <TouchableOpacity
                        key="gridItem5"
                        onLongPress={[Function]}
                        onPress={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "borderColor": "#000000",
                            "borderWidth": 2,
                            "flex": 0.3333333333333333,
                            "justifyContent": "center",
                            "margin": 0,
                          }
                        }
                      >
                        <react-native-mock>
                          <Text
                            style={
                              Object {
                                "color": "#000",
                                "fontSize": 60,
                              }
                            }
                          >
                            <react-native-mock>
                              H
                            </react-native-mock>
                          </Text>
                        </react-native-mock>
                      </TouchableOpacity>
                    </react-native-mock>
                  </View>
                  <View
                    key="678"
                    style={
                      Object {
                        "flex": 0.3333333333333333,
                        "flexDirection": "row",
                        "margin": 0,
                      }
                    }
                  >
                    <react-native-mock>
                      <TouchableOpacity
                        key="gridItem6"
                        onLongPress={[Function]}
                        onPress={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "borderColor": "#000000",
                            "borderWidth": 2,
                            "flex": 0.3333333333333333,
                            "justifyContent": "center",
                            "margin": 0,
                          }
                        }
                      >
                        <react-native-mock>
                          <Text
                            style={
                              Object {
                                "color": "#000",
                                "fontSize": 60,
                              }
                            }
                          >
                            <react-native-mock>
                              J
                            </react-native-mock>
                          </Text>
                        </react-native-mock>
                      </TouchableOpacity>
                      <TouchableOpacity
                        key="gridItem7"
                        onLongPress={[Function]}
                        onPress={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "borderColor": "#000000",
                            "borderWidth": 2,
                            "flex": 0.3333333333333333,
                            "justifyContent": "center",
                            "margin": 0,
                          }
                        }
                      >
                        <react-native-mock>
                          <Text
                            style={
                              Object {
                                "color": "#000",
                                "fontSize": 60,
                              }
                            }
                          >
                            <react-native-mock>
                              K
                            </react-native-mock>
                          </Text>
                        </react-native-mock>
                      </TouchableOpacity>
                      <TouchableOpacity
                        key="gridItem8"
                        onLongPress={[Function]}
                        onPress={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#ffa",
                            "borderColor": "#000000",
                            "borderWidth": 2,
                            "flex": 0.3333333333333333,
                            "justifyContent": "center",
                            "margin": 0,
                          }
                        }
                      >
                        <react-native-mock>
                          <Text
                            style={
                              Object {
                                "color": "#000",
                                "fontSize": 60,
                              }
                            }
                          >
                            <react-native-mock>
                              L
                            </react-native-mock>
                          </Text>
                        </react-native-mock>
                      </TouchableOpacity>
                    </react-native-mock>
                  </View>
                </react-native-mock>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "height": 50,
                    "justifyContent": "flex-start",
                    "marginHorizontal": 10,
                  }
                }
              >
                <react-native-mock>
                  <View
                    style={
                      Object {
                        "borderBottomColor": "#444",
                        "borderBottomWidth": 1,
                        "flex": 1,
                        "height": 30,
                        "marginLeft": 20,
                      }
                    }
                  >
                    <react-native-mock>
                      <Text
                        style={
                          Object {
                            "color": "#000",
                            "fontSize": 20,
                          }
                        }
                      >
                        <react-native-mock />
                      </Text>
                    </react-native-mock>
                  </View>
                  <View>
                    <react-native-mock>
                      <TouchableOpacity
                        onLongPress={[Function]}
                        onPress={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "borderRadius": 8,
                            "height": 50,
                            "justifyContent": "center",
                            "width": 60,
                          }
                        }
                      >
                        <react-native-mock>
                          <Icon
                            allowFontScaling={false}
                            name="md-backspace"
                            size={35}
                          >
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": undefined,
                                    "fontSize": 35,
                                  },
                                  undefined,
                                  Object {
                                    "fontFamily": "Ionicons",
                                    "fontStyle": "normal",
                                    "fontWeight": "normal",
                                  },
                                ]
                              }
                            >
                              <react-native-mock>
                                
                              </react-native-mock>
                            </Text>
                          </Icon>
                        </react-native-mock>
                      </TouchableOpacity>
                    </react-native-mock>
                  </View>
                </react-native-mock>
              </View>
            </react-native-mock>
          </ImageBackground>
        </react-native-mock>
      </View>
    </AnimatedComponent>
  </withAnimatable(View)>
</Grid>
`;
